core/sql/exp/ExpLOBaccess.cpp

```diff
@@ -1502,7 +1507,50 @@ Ex_Lob_Error ExLob::allocateDesc(ULng32 size, Int64 &descNum, Int64 &dataOffset,
     char logBuf[4096];
     lobDebugInfo("In ExLob::allocateDesc",0,__LINE__,lobTrace_);
     Int32 openFlags = O_RDONLY ;   
-    
+    if (size == 0) //we are trying to empty this lob.
+      {
+        //rename lob datafile
+        char * saveLobDataFile = new(getLobGlobalHeap()) char[MAX_LOB_FILE_NAME_LEN+6];
+        str_sprintf(saveLobDataFile, "%s_save",lobDataFile_);
+        Int32 rc2 = hdfsRename(fs_,lobDataFile_,saveLobDataFile);
+        if (rc2 == -1)
+          {
+            lobDebugInfo("Problem renaming datafile to save data file",0,__LINE__,lobTrace_);
+            NADELETEBASIC(saveLobDataFile,getLobGlobalHeap());
+            return LOB_DATA_FILE_WRITE_ERROR;
+          }
+        //create a new file of the same name.
+        hdfsFile fdNew = hdfsOpenFile(fs_, lobDataFile_,O_WRONLY|O_CREAT,0,0,0);
+        if (!fdNew) 
+          {
+            str_sprintf(logBuf,"Could not create/open file:%s",lobDataFile_);
+            lobDebugInfo(logBuf,0,__LINE__,lobTrace_);
+            
+            //restore previous version
+            Int32 rc2 = hdfsRename(fs_,saveLobDataFile,lobDataFile_);
+              if (rc2 == -1)
+                {
+                  lobDebugInfo("Problem restoring datafile . Will need to retry the update",0,__LINE__,lobTrace_);
+                  NADELETEBASIC(saveLobDataFile,getLobGlobalHeap());
+                  return LOB_DATA_FILE_WRITE_ERROR;
+                }
+            return LOB_DATA_FILE_OPEN_ERROR;
```

Comment:
its deleted below in the "else" part 
 else
          {
            //A new empty data file has been created.
            // delete the saved data file
            Int32 rc2 = hdfsDelete(fs_,saveLobDataFile,FALSE);//ok to ignore error.nt32            
            if (rc2 == -1)
              {
                lobDebugInfo("Problem deleting saved datafile . Will need to manually cleanup saved datafile",0,__LINE__,lobTrace_);
              }
            NADELETEBASIC(saveLobDataFile,getLobGlobalHeap());
            hdfsCloseFile(fs_,fdNew);
            fdNew = NULL;    
          }